File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        
        
        int d1 = common[2] - common[1];
        int d2 = common[1] - common[0];
            
        if(d1==d2){
            answer = common[common.length-1] + d1;
        }
        else{
            if(common[0] == 0) answer = 0;
            else{
                int r = common[1] / common[0];
                answer = common[common.length-1] * r;
            }
        }
        
        
        
        
        
        
        return answer;
    }
}

File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> common) {
    int answer = 0;
    
    int d1 = common[1] - common[0];
    int d2 = common[2] - common[1];
    
    if(d1 == d2){
        answer = common[common.size()-1] + d1;
    }
    else{
        answer = common[common.size()-1] * (common[1]/common[0]);
    }
    
    return answer;
}